File: Python/Functional-2/no_z.py

# given a list of strings, return a list of the strings, omitting any string that contains a 'z'
def no_z(strings):
    new_list = []

    for x in strings:
        if not 'z' in x:
            new_list.append(x)

    return new_list

print(no_z(['aaa', 'bbb', 'aza']))
print(no_z(['hziz', 'hzello', 'hi']))
print(no_z(['hello', 'howz', 'are', 'youz']))

File: JavaScript/Functional-2/noZ.js

// given a list of strings, return a list of the strings, omitting any string that contains a 'z'
const noZ = (strings) => {
    const newList = strings.filter(n => !n.includes('z'));

    return newList;
};

console.log(noZ(['aaa', 'bbb', 'aza']));
console.log(noZ(['hziz', 'hzello', 'hi']));
console.log(noZ(['hello', 'howz', 'are', 'youz']));
